### ISSUE TITLE
AUROC thresholds are persistent

### ISSUE THREAD

**Issue by GabrielDamsholt** [Lightning-AI/torchmetrics#1724]
## 🐛 Bug?

When using metrics from the AUROC family (https://torchmetrics.readthedocs.io/en/stable/classification/auroc.html) and setting the argument `thresholds` to a value different than `None`, a _persistent_ buffer is created under the name `thresholds` storing the precomputed tensor of thresholds in memory. As far as I can see the persistence of this buffer is unnecessary, and it is problematic in various scenarios outlined below.

### To Reproduce

Consider the simplified PyTorch Lightning workflow below in which an AUROC metric is tracked during _testing_ only, since the metric is too expensive to track during training. As shown one must use the keyword argument `strict = False` in order to test a fitted model.

```python
import pytorch_lightning as pl
from torchmetrics.classification import BinaryAUROC


class LightningModel(pl.LightningModule):
    def __init__(self):
        super().__init__()

    def setup(self, stage):
        if stage == "test":
            self.auroc_test = BinaryAUROC(thresholds=42)


model_train = LightningModel()
model_train.setup("train")
state_dict = model_train.state_dict()

model_test_1 = LightningModel()
model_test_1.setup("test")
model_test_1.load_state_dict(state_dict.copy(), strict=False)  # runs fine

model_test_2 = LightningModel()
model_test_2.setup("test")
model_test_2.load_state_dict(state_dict.copy(), strict=True)  # crashes
```

### Why This is a Bug/Problem

1. Adding the `strict=False` keyword argument seems hacky.
2. Alternatively one can initialize test metrics during training, but this seems wasteful and overly complicated.
3. When using PyTorch Lightning one can add the `strict=False` keyword argument to the `load_from_checkpoint` method, but this is not an option when using default checkpoint saving combined with the `ckpt_path` argument in `Trainer` (as well as in `LightningCLI`).
4. From a purely conceptual perspective it makes sense that `thresholds` should be non-persistent since
a. they remain constant after initialization,
b. initialization is cheap.
See the link in the Additional Context section below for further explanation of this.

Item 3 can be resolved by requesting the PyTorch Lightning team to add a `strict` option to checkpoint loading in their `Trainer`, but to me that solution seems incomplete since the root of the issue seems to be that AUROC thresholds should be non-persistent, as outlined in item 4.

</details>

### Solution

Simply make AUROC thresholds non-persistent by replacing the line
```python
self.register_buffer("thresholds", thresholds)
```
by the line
```python
self.register_buffer("thresholds", thresholds, persistent=False)
```
in lines 148, 314, and 493 in the file `torchmetrics/classification/precision_recall_curve.py`.
__I have verified that this breaks no tests.__

### Additional Context
See this resolved GitHub issue regarding the motivation behind non-persistent buffers: https://github.com/pytorch/pytorch/issues/18056.

### Additional Remarks
There are a few other places in `torchmetrics` where buffers are used. It may be worthwhile to consider whether these should be persistent or not, and document and/or test this.
